### PR TITLE
Add new resource for Chef 15

### PR DESCRIPTION
**Pull request by tas50**
This PR adds the new resources we shipped in Chef 15 and it also includes some property changes that we shipped in 15.

### CODE REVIEW

File: chef_master/source/resource_archive_file.rst [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+* ``action`` identifies which steps the chef-client will take to bring the node into the desired state.
+* ``destination``, ``group``, ``mode``, ``options``, ``overwrite``, ``owner``, and ``path`` are the properties available to this resource.
+
+Actions
+=====================================================
+
+The archive_file resource has the following actions:
+
+``:extract``
+    Description here.
```
Comment (mjingle) [inline chef_master/source/resource_archive_file.rst]: Missing description :)

File: chef_master/source/resource_archive_file.rst [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+   .. end_tag
+
+Properties
+=====================================================
+
+The archive_file resource has the following properties:
+
+``destination``
+   **Ruby Type:** String | ``REQUIRED``
```
Comment (mjingle) [inline chef_master/source/resource_archive_file.rst:55]: Is the location of this ``REQUIRED`` a new format? I was under the impression this type of information would be included in the description instead.

File: chef_master/source/resource_snap_package.rst [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+* ``action`` identifies which steps the chef-client will take to bring the node into the desired state.
+* ``channel``, ``options``, ``package_name``, ``source``, ``timeout``, and ``version`` are the properties available to this resource.
+
+Actions
+=====================================================
+
+The snap_package resource has the following actions:
+
+``:install``
+    Description here.
```
Comment (mjingle) [inline chef_master/source/resource_snap_package.rst]: There's a number of missing action descriptions in this section...

File: chef_master/source/resource_snap_package.rst [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+``not_if``
+  Prevent a resource from executing when the condition returns ``true``.
+
+``only_if``
+  Allow a resource to execute only if the condition returns ``true``.
+
+.. end_tag
+
+Examples
```
Comment (mjingle) [inline chef_master/source/resource_snap_package.rst:243]: Missing examples section?

File: chef_master/source/resource_windows_dfs_folder.rst [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+* ``name`` is the name given to the resource block.
+* ``action`` identifies which steps the chef-client will take to bring the node into the desired state.
+* ``description``, ``folder_path``, ``namespace_name``, and ``target_path`` are the properties available to this resource.
+
+Actions
+=====================================================
+
+The windows_dfs_folder resource has the following actions:
+
+``:delete``
```
Comment (mjingle) [inline chef_master/source/resource_windows_dfs_folder.rst:36]: Missing descriptions here.

File: chef_master/source/resource_archive_file.rst (hunk 2 above)
Comment (tas50) [reply chef_master/source/resource_archive_file.rst:55]: It's auto generated from resource data. We missed most of them previously when we relied on the author to put it in the description wording.

**Review by mjingle — APPROVED** [review]
tenor-26301596

**Pull request merged**